internal/logs/syslog_transport.go

```diff
@@ -0,0 +1,86 @@
+package logs
+
+import (
+	"errors"
+	"fmt"
+	"log/syslog"
+	"net"
+	"os"
+	"sync"
+	"time"
+)
+
+const (
+	priority = syslog.LOG_INFO
+)
+
+// We implement this way to force timestamp with the original ones.
+type syslogTransport struct {
+	server   string
+	protocol string
+	hostname string
+	conn     net.Conn
+	lock     sync.Mutex // guards conn
+}
+
+func NewSyslogServer(server string, protocol string) (*syslogTransport, error) {
+	proto := "tcp"
+	if protocol != "" {
+		proto = protocol
+	}
+	syslogInstance := &syslogTransport{
+		server:   server,
+		protocol: proto,
+		lock:     sync.Mutex{},
+	}
+	err := syslogInstance.Init()
+	if err != nil {
+		return nil, err
+	}
+	return syslogInstance, nil
+}
+
+func (s *syslogTransport) Init() error {
+	hostname, _ := os.Hostname()
+	var err error
+
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.hostname = hostname
+	s.conn, err = net.Dial(s.protocol, s.server)
```

Comment:
how many open connection do we expect to have? Is it a single connection per LogCollection that will be used by multiple writers per workload?

Reply:
One connection per each LogCollection, where multiple writers can write too.